### DISCUSSION TITLE
Unable to Find Payload information for Webhook Receivers.

### DISCUSSION THREAD

**Discussion by gileshinchcliff** [fluxcd/flux2#1146] [Q&A]
Hi Guys, 

I'm looking for information around webhook receivers. I want to be able to trigger an update off the back of the opening of a pull request and then follow a pipeline through multiple environments before finally deploying to Prod on Merge. 

So the only commit and source of truth happens in prod. 

I can see this could potentially be possible using webhook receivers, but although there seems to be information around setting them up with sites (ie bitbucket, github etc), I can't see what payload the receiver actually accepts anywhere (to be able to mimic it in CI if needs be).

A nice hint was in this comment: https://github.com/fluxcd/flux/issues/1128#issuecomment-516169601

But it's for an older version and doesn't really help with flux2.

**Comment by stefanprodan** [comment]
For the generic receiver the body can be anything e.g.:

```
curl <webhook-url> -X POST -d 'test'
```

Docs:
- Webhook guide: https://toolkit.fluxcd.io/guides/webhook-receivers/
- Generic receiver: https://toolkit.fluxcd.io/components/notification/receiver/#generic-hmac-receiver

For GitHub/GiLab/etc the payload is specific to each platform e.g. https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads

**Reply by gileshinchcliff** [reply]
Thanks @stefanprodan  So to clarify, we can't define the commit in the webhook? You just call it and flux will deploy whatever commit it thinks is most ready?

**Reply by stefanprodan** [reply]
Yes, you can’t bypass the GitRepository setting that would be against what flux is made for: gitops :)

**Reply by stefanprodan** [reply]
The webhooks have only one function, to speed up the sync so users don’t have to wait for the reconciliation interval.

**Reply by gileshinchcliff** [reply]
I was thinking less GitRepository, more having fine grained control of which commit triggers the deploy. ie: so long as you match the tag flux ignores it until the webhook gets hit. But if there are multiple commits which match the tag, which one does it choose?

That's why I was thinking we can pass the commit sha through the webhook.

**Reply by hiddeco** [reply]
> But if there are multiple commits which match the tag, which one does it choose?

How can a tag be tied to multiple commits, as a tag is static and points to a single commit?
